test/e2e/plugin_sync/plugin_sync_lifecycle_test.go

```diff
@@ -0,0 +1,370 @@
+			Expect(err).To(BeNil(), "should not get any error while generating CR files")
+
+			pluginWithIncorrectVersion := *pluginsToGenerateCRs[numberOfPluginsToInstall]
+			pluginWithIncorrectVersion.Version = pluginWithIncorrectVersion.Version + framework.RandomNumber(2)
+			pluginsWithIncorrectVer, pluginWithIncorrectVerCRFilePaths, err = framework.CreateTemporaryCRsForPluginsInGivenPluginGroup(append(make([]*framework.PluginInfo, 0), &pluginWithIncorrectVersion))
+			Expect(err).To(BeNil(), "should not get any error while generating CR files")
+
+			ApplyConfigOnKindCluster(tf, clusterInfo, pluginCRFilePaths)
+			ApplyConfigOnKindCluster(tf, clusterInfo, pluginWithIncorrectVerCRFilePaths)
+		})
+
+		// Test case: c. create context and make sure context has created
+		It("create context with kubeconfig and context", func() {
+			By("create context with kubeconfig and context")
+			contextName = "sync-e2e-" + framework.RandomString(4)
+			err := tf.ContextCmd.CreateConextWithKubeconfig(contextName, clusterInfo.KubeConfigPath, clusterInfo.ClusterKubeContext)
+			Expect(err).To(BeNil(), "context should create without any error")
+			active, err := tf.ContextCmd.GetActiveContext(string(types.TargetK8s))
+			Expect(err).To(BeNil(), "there should be a active context")
+			Expect(active).To(Equal(contextName), "the active context should be recently added context")
+		})
+		// Test case: d. list plugins and validate plugins info, make sure all plugins installed for which CR's has applied to KIND cluster and available in central repo
+		It("list plugins and validate plugins being installed after context being created", func() {
+			pluginsList, err = tf.PluginCmd.ListPluginsForGivenContext(contextName, true)
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(Equal(len(pluginsInfoForCRsApplied)), "number of plugins should be same as number of plugins CRs applied")
+			Expect(framework.CheckAllPluginsExists(pluginsList, pluginsInfoForCRsApplied)).Should(BeTrue(), " plugins being installed and plugins info for which CRs applied should be same")
+		})
+
+		// Test case: e. run plugin sync
+		It("Uninstall one of the installed plugin", func() {
+			_, err = tf.PluginCmd.Sync()
+			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf(framework.UnableToFindPluginForTarget, pluginsWithIncorrectVer[0].Name, pluginsWithIncorrectVer[0].Target)))
+
+			pluginsList, err = tf.PluginCmd.ListPluginsForGivenContext(contextName, true)
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(Equal(len(pluginsInfoForCRsApplied)), "number of plugins should be same as number of plugins CRs applied")
+			Expect(framework.CheckAllPluginsExists(pluginsList, pluginsInfoForCRsApplied)).Should(BeTrue(), "plugins being installed and plugins info for which CRs applied should be same")
+		})
```

Comment:
```
	// e. uninstall one of the installed plugin, make sure plugin is uninstalled,
	//		run plugin sync, make sure the uninstalled plugin has installed again.
```

The comment mentioned at the beginning of the `// Use case 3` and the tests written here don't match. We are not uninstalling the plugin before running sync.

Reply:
updated